CorsixTH/Lua/app.lua

```diff
@@ -744,8 +744,15 @@ function App:_loadLevel(level, difficulty, level_name, level_file, level_intro,
   self.map:setBlocks(self.gfx:loadSpriteTable("Data", "VBlk-0"))
   self.map:setDebugFont(self.gfx:loadFont("QData", "Font01V"))
 
+  local function determineFreeBuildMode()
+    local is_main_campaign = tonumber(new_map.level_number)
+    local is_custom_campaign = campaign_info ~= nil
+    return self.config.free_build_mode and not (is_main_campaign or is_custom_campaign)
+  end
+  local free_build_mode = determineFreeBuildMode()
+
   -- Load world
-  self.world = World(self)
+  self.world = World(self, free_build_mode)
```

Comment:
Make the function call directly here to be cleaner, as I made sure the function name was explanatory.

Reply:
> Make the function call directly here to be cleaner, as I made sure the function name was explanatory.

Ok. Applied.